Fix: open as text link on downloads now uses utf-8 encoding

ArneBab opened the pull request:
Tiny change, replaces most of the use-cases of the much more complex and incorrect https://github.com/hyphanet/fred/pull/970

Bombe committed:
⬆️ Update Mockito to latest 2.x version

Unfortunately, it currently looks like we won’t be able to update to a
newer version than 3, because Mockito starts using WeakReference as a
type for some internal representation or other, so we can’t mock objects
anymore that use it.

ArneBab committed:
Fix: open as text link on downloads now uses utf-8 encoding

Bombe commented on src/freenet/clients/http/FProxyToadlet.java:
```diff
@@ -949,7 +949,7 @@ public void run() {
 					// FIXME: is this safe? See bug #131
-					MediaType textMediaType = new MediaType("text/plain");
+					MediaType textMediaType = new MediaType("text/plain;charset=utf-8");
 					textMediaType.setParameter("charset", (e.getExpectedMimeType() != null) ? MediaType.getCharsetRobust(e.getExpectedMimeType()) : null);
```
It kinda looks like this line would immediately clobber whatever you’ve set in the constructor…

Bombe replied:
No, this would have been easier if you had written a test first. 😄 

(I’m currently trying to write one, but this would be easier if somebody merged https://github.com/hyphanet/fred/pull/1043 first, because that one contains a lot of shit for easier tests. 🙂)

ArneBab replied:
Do you really mean the "quote n2ntm PR"?

(it feels like the quoting PR still needs so much work that I should only finish it after the next release)

ArneBab committed:
Fix: set utf-8 as fallback, as it should have been.

Thanks to Bombe for the review!

ArneBab committed:
Merge remote-tracking branch 'torusrxxx/patch-3' into open-as-plain-text-utf8

Bombe committed:
💚 Exclude ancient Hamcrest dependency from JUnit

Bombe committed:
✅ Add matchers for verifying links in HTML documents

Bombe committed:
✅ Add possibility to use different translation for each test

Bombe committed:
✅ Add test implementation of ToadletContext

Bombe committed:
🐛 Set charset correctly

And add a test for it.

Bombe committed:
✅ Add matchers for TestToadletContext

Bombe committed:
✅ Add second test for FProxyToadlet

Bombe committed:
✅ Allow TestToadletContext to execute the request itself

Bombe committed:
✅ Add test for requesting an invalid key via parameter

Bombe committed:
✅ Add test for requesting valid key via parameter

Bombe committed:
✅ Add matcher for the HTML document of a toadlet context

Bombe committed:
✅ Add matchers for HTML

Bombe committed:
✅ Add test for / redirecting to welcome toadlet

Bombe committed:
✅ Add test for /favicon.ico redirect

Bombe committed:
✅ Add hasContentType matcher for TestToadletContext

Bombe committed:
✅ Add test for feed generation

Bombe committed:
✨ Allow activating robots file generation in context

Bombe committed:
✅ Add tests for /robots.txt

Bombe committed:
✅ Add tests for some permanent redirects

Bombe committed:
✅ Add test for requesting a failed download

Bombe committed:
✨ Add Document matcher that can locate Elements

Bombe committed:
✨ Add Document matcher that can locate and verify Elements

Bombe committed:
✨ Add Element matcher that can verify attributes

Bombe committed:
♻️ Use improved matchers

Bombe committed:
✅ Add test for the progress page

Bombe committed:
✅ Add more tests for redirects

ArneBab committed:
Merge remote-tracking branch 'Bombe/open-as-plain-text-utf8' into open-as-plain-text-utf8

ArneBab committed:
Add jsoup verification key

matches the key in https://github.com/apache/groovy/blob/master/gradle/verification-metadata.xml

Bombe committed:
🧑‍💻 Make test (slightly) less confusing

FProxyToadlet does not even reject anything, so the Accept headers are almost
completely useless; however, for a number of accepted MIME types we are
following redirects automatically, and that’s what this test is supposed to
verify.

ArneBab committed:
Merge remote-tracking branch 'Bombe/open-as-plain-text-utf8' into open-as-plain-text-utf8

Bombe committed:
⬆️ Use jakarta activation-api instead of relying on javax.activation

javax.activation was a part of the JDK until Java 9, when it was removed. By
using the new API (even though the old one is still available in Java 8) we
will remove that hurdle when switching to a newer version of Java.

ArneBab committed:
Merge remote-tracking branch 'Bombe/open-as-plain-text-utf8' into open-as-plain-text-utf8

ArneBab committed:
Refresh the verification metadata

ArneBab committed:
Merge branch 'next' into open-as-plain-text-utf8
